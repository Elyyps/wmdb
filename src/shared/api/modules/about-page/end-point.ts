import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
