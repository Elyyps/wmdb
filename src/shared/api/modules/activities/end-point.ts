import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
