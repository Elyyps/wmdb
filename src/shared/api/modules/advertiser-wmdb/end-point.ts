import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
