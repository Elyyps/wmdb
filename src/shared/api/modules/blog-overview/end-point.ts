import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
