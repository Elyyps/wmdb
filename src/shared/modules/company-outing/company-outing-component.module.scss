.company-outing {
  &__list {
    margin-bottom: 40px;
  }
  &__actions {
    text-align: center;
  }
  &__title {
    color: $global-inverse-color;
    margin: 0 0 56px;
    padding: 0 0 26px;
    position: relative;
    text-align: center;
    @media (max-width: $breakpoint-small-max) {
      margin: 0 0 32px;
      padding: 0 0 20px;
    }
    &::after {
      position: absolute;
      left: 50%;
      bottom: 0;
      content: "";
      width: 150px;
      height: 2px;
      background: currentColor;
      transform: translateX(-50%);
    }
  }
}
