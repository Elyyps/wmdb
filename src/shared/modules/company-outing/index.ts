export * from "./company-outing.component";
