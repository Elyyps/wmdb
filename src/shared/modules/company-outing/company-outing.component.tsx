import * as React from "react";
import styles from "./company-outing-component.module.scss";
import { ModuleSectionComponent } from "@app/core/module-section";
import { ICompanyOutingModule } from "@app/api/modules/company-outing/company-outing";
import { Button } from "@app/core/button";
import { CompanyOutingCardComponent } from "@app/core/company-outing-card";
export interface ICompanyOutingComponentProps {
  companyOutingModule: ICompanyOutingModule;
}

const CompanyOutingComponent = ({ companyOutingModule }: ICompanyOutingComponentProps) => (
  <ModuleSectionComponent
    title={companyOutingModule.title}
    titleColor={"white"}
    backgroundImage={companyOutingModule.backgroundImage}
    paddingTop={"120px"}
    paddingBottom={"120px"}
    visibility={"uk-visible@m"}
  >
    <div className={` ${styles["company-outing"]} `}>
      <div className="uk-container">
        {companyOutingModule.cards && (
          <div className={` ${styles["company-outing__list"]} uk-grid uk-child-width-1-3@s `}>
            {companyOutingModule.cards.map((item, key) => (
              <div key={key}>
                <CompanyOutingCardComponent {...item} />
              </div>
            ))}
          </div>
        )}
        <div className={styles["company-outing__actions"]}>
          <Button title={companyOutingModule.button.title} href={companyOutingModule.button.url} />
        </div>
      </div>
    </div>
  </ModuleSectionComponent>
);

export { CompanyOutingComponent };
