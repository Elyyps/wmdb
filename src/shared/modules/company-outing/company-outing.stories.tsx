import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { CompanyOutingComponent } from "./company-outing.component";
import { generateCompanyOutingData } from "@app/api/modules/company-outing/end-point";

storiesOf("CompanyOuting", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <CompanyOutingComponent companyOutingModule={generateCompanyOutingData()} />);
