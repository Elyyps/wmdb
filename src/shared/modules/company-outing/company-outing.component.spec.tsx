import React from "react";
import { shallow } from "enzyme";
import { CompanyOutingComponent } from "./company-outing.component";
import { generateCompanyOutingData } from "@app/api/modules/company-outing/end-point";

describe("[CompanyOuting]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<CompanyOutingComponent companyOuting={generateCompanyOutingData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
