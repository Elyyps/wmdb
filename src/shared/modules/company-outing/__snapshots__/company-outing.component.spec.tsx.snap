// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CompanyOuting] should render component without crashing 1`] = `
<ModuleSectionComponent
  backgroundImage="https://happytovisit.com/glib/3/Visit-Blue-Cave-and-5-Islands-with-Luxury-Speedboat-from-Split-2-19559.png"
  paddingBottom="120px"
  paddingTop="120px"
  title="Alles over het kiezen van een bedrijfsuitje"
  titleColor="white"
  visibility="uk-visible@m"
>
  <div
    className=" company-outing "
  >
    <div
      className="uk-container"
    >
      <div
        className=" company-outing__list uk-grid uk-child-width-1-3@s "
      >
        <div
          key="0"
        >
          <CompanyOutingCardComponent
            button={
              Object {
                "title": "Lees meer",
                "url": "#",
              }
            }
            content="Lorem ipsum dolor sit amet, consectetu dipiscing elit. Duis quis tortor lectus. Aliquam vel dignissim nunc…"
            data="18 sep. 2017"
            icon="eten.svg"
            image="https://picsum.photos/id/116/300/200"
            title="Blog titel"
          />
        </div>
        <div
          key="1"
        >
          <CompanyOutingCardComponent
            button={
              Object {
                "title": "Lees meer",
                "url": "#",
              }
            }
            content="Lorem ipsum dolor sit amet, consectetu dipiscing elit. Duis quis tortor lectus. Aliquam vel dignissim nunc…"
            data="18 sep. 2017"
            icon="eten.svg"
            image="https://picsum.photos/id/118/300/200"
            title="Blog titel"
          />
        </div>
        <div
          key="2"
        >
          <CompanyOutingCardComponent
            button={
              Object {
                "title": "Lees meer",
                "url": "#",
              }
            }
            content="Lorem ipsum dolor sit amet, consectetu dipiscing elit. Duis quis tortor lectus. Aliquam vel dignissim nunc…"
            data="18 sep. 2017"
            icon="eten.svg"
            image="https://picsum.photos/id/115/300/200"
            title="Blog titel"
          />
        </div>
      </div>
      <div
        className="company-outing__actions"
      >
        <Button
          href="#"
          title="Bekijk álle blogs"
        />
      </div>
    </div>
  </div>
</ModuleSectionComponent>
`;
